src/vivarium_gates_mncnh/components/antenatal_care.py

```diff
@@ -0,0 +1,292 @@
+        )
+
+    def on_time_step(self, event: Event) -> None:
+        if self._sim_step_name() == SIMULATION_EVENT_NAMES.PREGNANCY:
+            self.transition(event.index, event.time)
+
+
+class DecisionTreeState(State):
+    def setup(self, builder: Builder) -> None:
+        self._sim_step_name = builder.time.simulation_event_name()
+        self.location = get_location(builder)
+
+    def add_decision(
+        self,
+        output_state: State,
+        decision_function: Callable[[pd.Index], pd.Series],
+    ) -> None:
+        transition = Transition(
+            self, output_state, self.get_probability_function(decision_function)
+        )
+        self.add_transition(transition)
+
+    def get_probability_function(
+        self, decision_function: Callable[[pd.Index], pd.Series]
+    ) -> Callable[[pd.Index], pd.Series]:
+        """We need to check the simulation step name within the probability function, because it will change"""
+
+        def probability_function(index: pd.Index) -> pd.Series:
+            if self._sim_step_name() != SIMULATION_EVENT_NAMES.PREGNANCY:
+                return lambda index: pd.Series(0.0, index=index)
+            return decision_function(index)
+
+        return probability_function
+
+
+class TransientDecisionTreeState(DecisionTreeState, Transient):
+    pass
+
+
+class ANCState(TransientDecisionTreeState):
+    def __init__(self) -> None:
+        super().__init__("attended_antental_care")
+
+    @property
+    def columns_required(self) -> list[str]:
+        return [COLUMNS.ATTENDED_CARE_FACILITY]
+
+    def transition_side_effect(self, index: pd.Index, _event_time: ClockTime) -> None:
+        pop = self.population_view.get(index)
+        pop[COLUMNS.ATTENDED_CARE_FACILITY] = True
+        self.population_view.update(pop)
+
+
+class StandardUltrasound(TransientDecisionTreeState):
+    def __init__(self) -> None:
+        super().__init__("standard_ultasound")
+
+    @property
+    def columns_required(self) -> list[str]:
+        return [COLUMNS.ULTRASOUND_TYPE]
+
+    def transition_side_effect(self, index: pd.Index, _event_time: ClockTime) -> None:
+        pop = self.population_view.get(index)
+        pop[COLUMNS.ULTRASOUND_TYPE] = ULTRASOUND_TYPES.STANDARD
+        self.population_view.update(pop)
+
+
+class AIAssistedUltrasound(TransientDecisionTreeState):
+    def __init__(self) -> None:
+        super().__init__("ai_assisted_ultrasound")
+
+    @property
+    def columns_required(self) -> list[str]:
+        return [COLUMNS.ULTRASOUND_TYPE]
+
+    def transition_side_effect(self, index: pd.Index, _event_time: ClockTime) -> None:
+        pop = self.population_view.get(index)
+        pop[COLUMNS.ULTRASOUND_TYPE] = ULTRASOUND_TYPES.AI_ASSISTED
+        self.population_view.update(pop)
+
+
+def ANC() -> Machine:
+    initial_state = DecisionTreeState("initial")
+    attended_antental_care = ANCState()
+    gets_ultrasound = TransientDecisionTreeState("gets_ultrasound")
```

Comment:
Addressing your last two comments here - we are subclassing TransientDecisionTreeState for each of the states so I didn't find it necessary to add transient to all the names since they are all transient states except the first and last states of the tree.

The string is a required arg for the State class but in the classes such as ANCSTATE() we are providing that inside the class's constructor.